### PR TITLE
Alexmont/debug view

### PR DESCRIPTION
**Pull request by alexmontAmazon**
debug viewer for Dom adapters, primarily used for internal testing while we get the full DPE view up and running.

### CODE REVIEW

File: Code/Framework/AzQtComponents/AzQtComponents/DPEDebugViewStandalone/main.cpp [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    DPEDebugView::DPEDebugApplication app(&argc, &argv);
+    AZ::IO::FixedMaxPath engineRootPath;
+    {
+        auto settingsRegistry = AZ::SettingsRegistry::Get();
+        settingsRegistry->Get(engineRootPath.Native(), AZ::SettingsRegistryMergeUtils::FilePathKey_EngineRootFolder);
+    }
+    AzQtComponents::PrepareQtPaths();
+    QCoreApplication::setAttribute(Qt::AA_EnableHighDpiScaling);
+    QCoreApplication::setAttribute(Qt::AA_UseHighDpiPixmaps);
+    QGuiApplication::setHighDpiScaleFactorRoundingPolicy(Qt::HighDpiScaleFactorRoundingPolicy::PassThrough);
```
Comment (cgalvan) [inline Code/Framework/AzQtComponents/AzQtComponents/DPEDebugViewStandalone/main.cpp:74]: Nit: You can use the `AzQtApplication` as your `QApplication` to get rid of some of this boilerplate

**Review by cgalvan — APPROVED** [review]
A screenshot would be nice so we can see it in action :)  But overall, looks good!

File: Code/Framework/AzToolsFramework/AzToolsFramework/UI/DPEDebugViewer/DPEDebugModel.cpp [hunk 2]
```diff
@@ -0,0 +1,374 @@
...
+        m_adapter = theAdapter;
+        m_isResetting = (m_rootNode != nullptr);
+        if (m_isResetting)
+        {
+            beginResetModel();
+            m_rootNode->deleteLater(); // recursively destroys the existing tree
+        }
+
+        // recursively populate the node tree with the new adapter information
+        auto theContents = theAdapter->GetContents();
```
Comment (NicholasVanSickle) [inline Code/Framework/AzToolsFramework/AzToolsFramework/UI/DPEDebugViewer/DPEDebugModel.cpp:233]: nit: looks like this is unused

File: Code/Framework/AzToolsFramework/AzToolsFramework/UI/DPEDebugViewer/DPEDebugModel.cpp [hunk 3]
```diff
@@ -0,0 +1,374 @@
...
+        if (m_isResetting)
+        {
+            endResetModel();
+            m_isResetting = false;
+        }
+        else
+        {
+            // inform any listening views that top-level rows have been added,
+            // they will automatically traverse the child nodes
+            const int numRootChildren = m_rootNode->GetChildCount();
```
Comment (NicholasVanSickle) [inline Code/Framework/AzToolsFramework/AzToolsFramework/UI/DPEDebugViewer/DPEDebugModel.cpp:246]: What does avoiding the model reset here get us? I've generally always done a reset model on populate, figuring the view has to query the whole structure anyways, but I've never actually profiled this, just seems like maybe we could get rid of `m_isResetting` entirely and streamline this.

File: Code/Framework/AzToolsFramework/AzToolsFramework/UI/DPEDebugViewer/DPEDebugModel.cpp [hunk 4]
```diff
@@ -0,0 +1,374 @@
...
+        if (m_isResetting)
+        {
+            beginResetModel();
+            m_rootNode->deleteLater(); // recursively destroys the existing tree
+        }
+
+        // recursively populate the node tree with the new adapter information
+        auto theContents = theAdapter->GetContents();
+        m_rootNode = new DPEModelNode(DPEModelNode::NodeType::RootNode, 0, this);
+        m_rootNode->Populate(theAdapter->GetContents());
```
Comment (NicholasVanSickle) [inline Code/Framework/AzToolsFramework/AzToolsFramework/UI/DPEDebugViewer/DPEDebugModel.cpp:235]: Not for this PR, but for anyone interested this is where we can wire up `ConnectResetHandler` and `ConnectChangedHandler`. For our debug view, resetting the model for both might be in "good enough" territory.

**Review by NicholasVanSickle — APPROVED** [review]
Looks good to me, excited to see this go in! Have a bit of feedback, nothing blocking.

File: Code/Framework/AzQtComponents/AzQtComponents/DPEDebugViewStandalone/main.cpp (hunk 1 above)
Comment (alexmontAmazon) [reply Code/Framework/AzQtComponents/AzQtComponents/DPEDebugViewStandalone/main.cpp:74]: I just tried this but it broke more than it fixed, so I'm cowardly reverting to the AzToolsFramework version :-/

File: Code/Framework/AzToolsFramework/AzToolsFramework/UI/DPEDebugViewer/DPEDebugModel.cpp (hunk 2 above)
Comment (alexmontAmazon) [reply Code/Framework/AzToolsFramework/AzToolsFramework/UI/DPEDebugViewer/DPEDebugModel.cpp]: fixed, thanks!

**Comment by alexmontAmazon** [comment]
now with cool screenshot!
image

**Pull request merged**